community/io/src/main/java/org/neo4j/io/pagecache/impl/muninn/PageList.java

```diff
@@ -71,6 +70,22 @@
         clearMemory( baseAddress, pageCount );
     }
 
+    /**
+     * This copy-constructor is useful for classes that want to extend the {@code PageList} class to inline its fields.
+     * All data and state will be shared between this and the given {@code PageList}. This means that changing the page
+     * list state through one has the same effect as changing it through the other – they are both effectively the same
+     * object.
+     * @param pageList The {@code PageList} instance whose state to copy.
+     */
+    PageList( PageList pageList )
```

Comment:
It's so `MuninnPagedFile` can extend `PageList` itself, and get all the relevant state carried over from the page list instance created in `MuninnPageCache`.